src/font_fallback.rs

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::cell::UnsafeCell;
+use std::ptr::null_mut;
+
+use winapi::um::dwrite_2::{IDWriteFactory2, IDWriteFontFallback};
+
+use super::*;
+use helpers::*;
+
+pub struct FontFallback {
+    native: UnsafeCell<ComPtr<IDWriteFontFallback>>,
+}
+
+pub struct FallbackResult {
+    /// Length of mapped substring, in utf-16 code units.
+    pub mapped_length: usize,
+    /// The font that should be used to render the substring.
+    pub mapped_font: Option<Font>,
+    /// The scale factor to apply.
+    pub scale: f32,
+}
+
+impl FontFallback {
+    pub fn get_system_fallback() -> Option<FontFallback> {
+        unsafe {
+            let factory = ComPtr::already_addrefed(DWriteFactory());
+            let factory2 = factory.query_interface::<IDWriteFactory2>(&IDWriteFactory2::uuidof());
+            std::mem::forget(factory);
+            let factory2 = factory2?;
+            let mut native = null_mut();
+            let hr = factory2.GetSystemFontFallback(&mut native);
+            assert!(hr == 0);
```

Comment:
nit: could be `assert_eq!()`, that way it'll dump the error code if it fails

Reply:
I'm following crate conventions here (the pattern is used extensively). I'm not a fan in any case - it's panic-prone when things do go wrong. Do you want me to use assert_eq in my new code only (which would make it inconsistent with rest of crate), or change all instances? Or leave it alone?

Reply:
Ah, I see. I guess using `assert_eq!` in new code only is fine.